src/video-uploader.ts

```diff
@@ -42,10 +44,21 @@ export class VideoUploader extends AbstractUploader<UploadProgressEvent> {
         this.fileName = options.videoName || this.file.name;
 
         this.chunksCount = Math.ceil(this.fileSize / this.chunkSize);
+        this.maxVideoDuration = options.maxVideoDuration;
     }
 
 
     public async upload(): Promise<VideoUploadResponse> {
+        if (this.maxVideoDuration !== null && !document) {
+            throw Error('document is undefined. Impossible to use the maxVideoDuration option. Remove it and try again.')
+        }
+        let tooLong: boolean = false;
+        if (this.maxVideoDuration !== null) {
+            tooLong = await this.checkVideoDuration();
+        }
+        if (tooLong) {
+            throw Error(`The video submitted is too long.`);
+        }
```

Comment:
Could be simpler by renaming `checkVideoDuration` by ìsVideoTooLong` and writting something like:
```js
if (this.maxVideoDuration !== undefined && await this.isVideoTooLong()) {
   throw Error(`The video submitted is too long.`);
}
```